File: Java/Calculator/src/main/java/com/softuni/entity/Calculator.java
package com.softuni.entity;

public class Calculator { // nqma nujda ot pisane na nqkakvu putishta ili kakvoto i da bilo. Tova si e logikata, koqto shte se polzva ot controller-a
    private double leftOperand;
    private double rightOperand;
    private String operator;

    public double getLeftOperand() {
        return this.leftOperand;
    }

    public double getRightOperand() {
        return this.rightOperand;
    }

    public String getOperator() {
        return this.operator;
    }

    public Calculator(double leftOperand, double rightOperand, String operator) {
        this.leftOperand = leftOperand;
        this.rightOperand = rightOperand;
        this.operator = operator;
    }

    public double calculateResult() {
        double result = 0.0;

        switch (this.operator) {
            case "+":
                result = this.leftOperand + this.rightOperand;
                break;

            case "-":
                result = this.leftOperand - this.rightOperand;
                break;

            case "*":
                result = this.leftOperand * this.rightOperand;
                break;

            case "/":
                result = this.leftOperand / this.rightOperand;
                break;

            case "^":
                result = Math.pow(this.leftOperand, this.rightOperand);
                break;

            case "root":
                result = Math.pow(this.leftOperand, 1.0 / this.rightOperand);
                break;
        }

        return result;
    }
}

File: Csharp/Calculator/CalculatorApp/Models/Calculator.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CalculatorApp.Models
{
    public class Calculator
    {
        public Calculator()
        {
            this.Result = 0.0;
        }

        public double LeftOperand { get; set; }
        public string Operator { get; set; }
        public double RightOperand { get; set; }
        public double Result { get; set; }
        public string LastOperation { get; set; }

        public void CalculateResult()
        {
            switch (this.Operator)
            {
                case "+":
                    this.Result = this.LeftOperand + this.RightOperand;
                    break;
                case "-":
                    this.Result = this.LeftOperand - this.RightOperand;
                    break;
                case "*":
                    this.Result = this.LeftOperand * this.RightOperand;
                    break;
                case "/":
                    this.Result = this.LeftOperand / this.RightOperand;
                    break;
                case "^":
                    this.Result = Math.Pow(this.LeftOperand, this.RightOperand);
                    break;
                case "√":
                    this.Result = Math.Pow(this.LeftOperand, 1.0/this.RightOperand);
                    break;
            }
        }
        public void BuildLastOperation()
        {
            this.LastOperation = $"{this.LeftOperand} {this.Operator} {this.RightOperand} = {this.Result}";
        }


    }
}
